intentgen/src/main/java/com/chatbot/IntentGenerator.java

```diff
@@ -0,0 +1,184 @@
+
+import com.chatbot.protobuf.UseCaseOuterClass.UseCase;
+import com.google.cloud.dialogflow.v2.Intent;
+import com.google.cloud.dialogflow.v2.Context;
+import com.google.cloud.dialogflow.v2.Intent.TrainingPhrase;
+import com.google.cloud.dialogflow.v2.Intent.TrainingPhrase.Part;
+import com.google.cloud.dialogflow.v2.IntentsClient;
+import com.google.cloud.dialogflow.v2.ProjectAgentName;
+import com.google.protobuf.TextFormat;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+public class IntentGenerator {
+  static final List<String> YES_TRAINING_PHRASES = new ArrayList<String>(Arrays.asList("yes",
+      "okay I will", "why not", "yes that's alright", "yes I do", "exactly", "of course",
+      "yep that's ok", "okay", "ok", "sure"));
+  static final List<String> NO_TRAINING_PHRASES = new ArrayList<String>(Arrays.asList("No",
+      "thanks but no", "no way", "no no don't", "na", "no it isn't", "don't", "nah I'm good",
+      "no I cannot", "I can't"));
+  private static Map<String, String> intentDisplayNameToName;
+
+  public static void main(final String[] args) throws ParseException {
+    Options options = new Options();
+    final Option useCaseFile = Option.builder("f").hasArg()
+        .desc("prototxt file describing the use case" ).build();
+    options.addOption(useCaseFile);
+    final CommandLineParser parser = new DefaultParser();
+    final CommandLine cmd = parser.parse(options, args);
+    final String fileName = cmd.getOptionValue("f");
+    if(fileName.isEmpty()) {
+      throw new ParseException("No file name provided");
+    }
+    final String projectID = System.getenv("projectID");
+    final ProjectAgentName parent = ProjectAgentName.of(projectID);
+    try(IntentsClient intentsClient = IntentsClient.create()) {
+      // this map is needed to check if an intent display name already exists and if so, instead of 
+      // creating an intent (which would return an error) we update the intent using the name
+      intentDisplayNameToName = getIntentDisplayNametoNameMapping(intentsClient, parent);
+      final List<Intent> intentProtobufList = getIntentsInUseCase(fileName).stream()
+          .map(x->buildIntentProtobufForIntent(x, projectID))
+          .collect(Collectors.toList());
+      createOrUpdateIntents(intentProtobufList, intentsClient, intentDisplayNameToName, parent);
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+  }
+
+  // To include entities into strings the syntax used is: |entity-type;alias;value|
+  // A valid string would be "some text |@sys.name;name;Foo| some text"
+  public static TrainingPhrase buildTrainingPhraseFromEncodedString(String trainingPhrase) {
+    int ENTITY_TYPE_INDEX = 0;
+    int ALIAS_INDEX = 1;
+    int VALUE_INDEX = 2;
```

Comment:
put these as static class members or. somewhere that is also accessible in the test